File: presentations/async-await-laziness-in-detail/futures-vs-promises.js
async function subtask() {
  console.log("> > subtask"); // <3>
}

async function task() {
  console.log("> before subtask"); // <2>
  await subtask();
  console.log("> after subtask"); // <5>
}

(async function main() {
  console.log("before promise"); // <1>
  let promise = task();
  console.log("promise is created"); // <4>
  await promise;
  console.log("promise is awaited"); // <6>
})();

File: presentations/async-await-laziness-in-detail/futures_vs_promises.rs
async fn subtask() {
    println!("> > subtask"); // <4>
}

async fn task() {
    println!("> Before subtask"); // <3>
    subtask().await;
    println!("> After subtask"); // <5>
}

fn main() {
    futures::executor::block_on(async {
        println!("before future"); // <1>
        let future = task();
        println!("future is created"); // <2>
        future.await;
        println!("future is awaited"); // <6>
    });
}
